src/main/java/com/google/sps/servlets/BusinessDataServlet.java

```diff
@@ -0,0 +1,207 @@
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import com.google.gson.Gson;
+import java.lang.reflect.Type;
+import com.google.gson.reflect.TypeToken;
+import com.google.sps.Business;
+import com.google.sps.Constants;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.text.SimpleDateFormat;  
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns all businesses and lets you create a new business */
+@WebServlet("/business-data")
+public class BusinessDataServlet extends HttpServlet {
+    
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Create gson for serializing and deserializing
+    Gson gson = new Gson();
+
+    // Create query for Datastore.
+    Query query = new Query("Business");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Iterate over results and add each business to the ArrayList.
+    List<Business> businesses = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty(Constants.BUSINESS_NAME);
+      String desc = (String) entity.getProperty(Constants.BUSINESS_DESC);
+      String[] categoriesArr = gson.fromJson((String) entity.getProperty(Constants.BUSINESS_CATEGORIES), String[].class);
+      List<String> categories = Arrays.asList(categoriesArr);
+      String address = (String) entity.getProperty(Constants.BUSINESS_ADDRESS);
+      String addressLatStr = (String) entity.getProperty(Constants.BUSINESS_ADDRESS_LAT);
+      String addressLngStr = (String) entity.getProperty(Constants.BUSINESS_ADDRESS_LNG);
+      float addressLat;
+      if (addressLatStr.isEmpty()) {
+          addressLat = 404;
+      } else {
+          addressLat = Float.parseFloat(addressLatStr);
```

Comment:
Why not store it as a float in the first place instead of doing the conversion afterwards?

Reply:
Yup you're right. I've changed it accordingly.